src/clj/puppetlabs/services/protocols/jruby_pool.clj

```diff
@@ -25,14 +25,15 @@
 
   (borrow
     [pool-context]
-    "Returns a reference to a JRuby instance. Will block if the pool is locked or no instances
-    are available.")
+    "Returns a reference to a JRuby instance and a worker id (instance id or thread id).
+    Will block if the pool is locked or no instances are available.")
 
 
   (borrow-with-timeout
     [pool-context timeout]
-    "Returns a reference to a JRuby instance. Will block if the pool is locked or no instances
-    are available, timing out when the supplied number of milliseconds has elapsed.")
+    "Returns a reference to a JRuby instance and a worker id (instance id or thread id).
+    Will block if the pool is locked or no instances are available, timing out when the
+    supplied number of milliseconds has elapsed.")
 
   (return
```

Comment:
Can you add something to the doc string here that this will now return the id of the thing being returned?